consensus/istanbul/backend/snapshot.go

```diff
@@ -70,6 +70,7 @@ func loadSnapshot(epoch uint64, db ethdb.Database, hash common.Hash) (*Snapshot,
 
 // store inserts the snapshot into the database.
 func (s *Snapshot) store(db ethdb.Database) error {
+	s.ValSet.CacheUncompressed()
```

Comment:
CacheUncompressed() rigth now computes the uncompressed key wether it's already set or not... which it should. It should only compute it when it's not already cached